test/unit/pusher_test.rb

```diff
@@ -263,16 +263,11 @@ class PusherTest < ActiveSupport::TestCase
       @cutter.stubs(:rubygem).returns @rubygem
       create(:version, rubygem: @rubygem, number: '0.1.1', summary: 'old summary')
       @cutter.stubs(:version).returns @rubygem.versions[0]
-      @rubygem.stubs(:update_attributes_from_gem_specification!)
-      GemCachePurger.stubs(:call)
+      @rubygem.expects(:update_attributes_from_gem_specification!).with(@cutter.version, nil).at_least_once
+      GemCachePurger.expects(:call).with(@rubygem.name).at_least_once
       Indexer.any_instance.stubs(:write_gem)
-      @cutter.save
-    end
 
-    should "update rubygem attributes" do
```

Comment:
We can't delete this. Even when the code technically still tests the same thing, this `should update rubygem attributes` is important spec/information we can't lose. I understand `stubs` were originally in `setup` but now we need to add `@rubygem.expects..` in `should` block and then call the function we are testing. It will inevitably lead to some code duplication and that is fine.

Same for all other places you have removed a test description. Removing/Improving not so useful one like `call GemCachePurger` is acceptable.

Reply:
Updated to preserve what seemed like meaningful tests, removed a few one-liners.